### PR TITLE
Reduce the default batch size.

### PR DESCRIPTION
**Pull request by PavelZaytsev**
## Overview

Description:
- Reduce the default read batch size for transfer operations to address the out of direct memory issue for Netty Server Router.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java [hunk 1]
```diff
@@ -105,7 +105,7 @@
                     + " -k <seqcache>, --sequencer-cache-size=<seqcache>                         "
                     + "               The size of the sequencer's cache. [default: 250000].\n    "
                     + " -B <size> --batch-size=<size>                                            "
-                    + "              The read/write batch size used for data transfer operations [default: 100].\n"
+                    + "              The read/write batch size used for data transfer operations [default: 20].\n"
```
Comment (WenbinZhu) [inline infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java:108]: LR also has a CorfuServer, called `CorfuInterClusterReplicationServer`, which is mostly a copy of this one and also has default batch size of 100, probably need to take care of that one as well.

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java [hunk 2]
```diff
@@ -32,10 +32,23 @@
     /**
      * A number of in-flight requests per one node.
      */
-    private static final int NUM_REQUESTS_PER_NODE = 5;
+    private final int batchesPerNode;
```
Comment (WenbinZhu) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java:35]: This not really a batch, it's the number of concurrent requests, we need to change the name.

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java [hunk 3]
```diff
@@ -32,10 +32,23 @@
...
     public ParallelTransferProcessor(StateTransferBatchProcessor stateTransferBatchProcessor) {
         this.stateTransferBatchProcessor = stateTransferBatchProcessor;
+        this.batchesPerNode = 5;
+    }
+
+    public ParallelTransferProcessor(StateTransferBatchProcessor stateTransferBatchProcessor,
+                                     int batchSize) {
+        this.stateTransferBatchProcessor = stateTransferBatchProcessor;
+        verifyBatchSize(batchSize);
+        this.batchesPerNode = DEFAULT_THROUGHPUT_PER_NODE / batchSize;
```
Comment (WenbinZhu) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java:51]: Going by this calculation, there will be 500/20=25 concurrent requests for a single node, which I think maybe too large. I guess you did this to match the existing throughput, but I think we might not be able to make it the same when reducing the batchsize.

**Review by xnull — APPROVED** [review]
looks ok

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java (hunk 3 above)
Comment (PavelZaytsev) [reply infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java]: fixed

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java (hunk 2 above)
Comment (PavelZaytsev) [reply infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java]: fixed

File: infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java (hunk 1 above)
Comment (PavelZaytsev) [reply infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java:108]: fixed

File: infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java [hunk 4]
```diff
@@ -24,6 +24,10 @@
  */
 @Slf4j
 public class ParallelTransferProcessor {
+    /**
+     * Default number of in-flight requests per node.
+     */
+    private static final int DEFAULT_NUM_INFLIGHT_REQUESTS = 5;
```
Comment (Maithem) [inline infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java:30]: I know its tedious but can you make this configurable ?

**Pull request closed without merging**